compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -413,6 +472,81 @@ void RecordMinMax::profileData(const std::string &mode, const std::string &input
   update_quantparam(getObserver(), mode, min_percentile, max_percentile);
 }
 
+void RecordMinMax::profileDataInParallel(const std::string &mode,
+                                         const std::string &input_data_path, float min_percentile,
+                                         float max_percentile)
+{
+  assert(_interpreters.size() == _threads_size);
+  assert(_observers.size() == _threads_size);
+  const auto vector_input_data = importH5Data(input_data_path);
+  const auto num_records = vector_input_data.size();
+  const auto input_nodes = loco::input_nodes(_module->graph());
+
+  // Start parallel part
+  std::cout << _threads_size << " concurrent threads are supported.\n";
+
+  if (num_records < _threads_size)
+  {
+    _threads_size = num_records;
+  }
```

Comment:
as I understand, this method runs and exits the problem and this code is OK.
but I think using local variable for this would be better for future changes.
```suggestion
  auto run_threads = num_records < _threads_size ? num_records : _threads_size 
```